File: FCL/src/main/java/com/tungsten/fcl/ui/controller/ControllerPageManager.java
package com.tungsten.fcl.ui.controller;

import android.content.Context;

import com.tungsten.fcl.R;
import com.tungsten.fcl.ui.PageManager;
import com.tungsten.fcl.ui.UIListener;
import com.tungsten.fcllibrary.component.ui.FCLCommonPage;
import com.tungsten.fcllibrary.component.view.FCLUILayout;

import java.util.ArrayList;

public class ControllerPageManager extends PageManager {

    public static final int PAGE_ID_CONTROLLER_MANAGER = 15040;
    public static final int PAGE_ID_CONTROLLER_REPO = 15041;

    private static ControllerPageManager instance;

    private ControllerManagePage controllerManagePage;
    private ControllerRepoPage controllerRepoPage;

    public static ControllerPageManager getInstance() {
        if (instance == null) {
            throw new IllegalStateException("ControllerPageManager not initialized!");
        }
        return instance;
    }

    public ControllerPageManager(Context context, FCLUILayout parent, int defaultPageId, UIListener listener) {
        super(context, parent, defaultPageId, listener);
        instance = this;
    }

    @Override
    public void init(UIListener listener) {
        controllerManagePage = new ControllerManagePage(getContext(), PAGE_ID_CONTROLLER_MANAGER, getParent(), R.layout.page_controller_manager);
        controllerRepoPage = new ControllerRepoPage(getContext(), PAGE_ID_CONTROLLER_REPO, getParent(), R.layout.page_controller_repo);

        if (listener != null) {
            listener.onLoad();
        }
    }

    @Override
    public ArrayList<FCLCommonPage> getAllPages() {
        ArrayList<FCLCommonPage> pages = new ArrayList<>();
        pages.add(controllerManagePage);
        pages.add(controllerRepoPage);
        return pages;
    }
}

File: FCL/src/main/java/com/tungsten/fcl/ui/controller/ControllerPageManager.kt
package com.tungsten.fcl.ui.controller

import android.content.Context
import com.tungsten.fcl.R
import com.tungsten.fcl.ui.PageManager
import com.tungsten.fcllibrary.component.ui.FCLCommonPage
import com.tungsten.fcllibrary.component.view.FCLUILayout

class ControllerPageManager(
    context: Context?,
    parent: FCLUILayout?,
    defaultPageId: Int
) : PageManager(context, parent, defaultPageId) {
    companion object {
        const val PAGE_ID_CONTROLLER_MANAGER: Int = 15040
        const val PAGE_ID_CONTROLLER_REPO: Int = 15041

        @JvmStatic
        var instance: ControllerPageManager? = null
    }

    private lateinit var controllerManagePage: ControllerManagePage
    private val controllerRepoPage: ControllerRepoPage by lazy {
        ControllerRepoPage(
            context,
            PAGE_ID_CONTROLLER_REPO,
            parent,
            R.layout.page_controller_repo
        )
    }

    init {
        instance = this
    }

    override fun init() {
        controllerManagePage = ControllerManagePage(
            context,
            PAGE_ID_CONTROLLER_MANAGER,
            parent,
            R.layout.page_controller_manager
        )
    }

    override fun getAllPages(): ArrayList<FCLCommonPage> {
        return ArrayList<FCLCommonPage>().apply {
            add(controllerManagePage)
        }
    }

    override fun createPageById(id: Int): FCLCommonPage? {
        val page: FCLCommonPage? = when (id) {
            PAGE_ID_CONTROLLER_REPO -> controllerRepoPage
            else -> null
        }
        if (page != null) {
            allPages.add(page)
        }
        return page
    }
}
